druid/src/app_delegate.rs

```diff
@@ -56,13 +48,24 @@ impl<'a, 'b> DelegateCtx<'a, 'b> {
 ///
 /// You customize the `AppDelegate` by passing closures during creation.
 pub struct AppDelegate<T> {
-    event_fn: Option<Box<dyn Fn(Event, &mut T, &Env, &mut DelegateCtx) -> Option<Event> + 'static>>,
+    event_fn: Option<
```

Comment:
I agree that this complexity isn't worth it, and we should just back up and make `AppDelegate` a trait.

If you'd like to try that, it's probably a separate PR, and then we could redo this on top of that? Or I could make that change, whichever is easiest.

Reply:
Yeah, I'd like to try my hand at that. I'll leave this as-is for now then.